src/Symfony/Component/ErrorHandler/Resources/bin/patch-type-declarations.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+if (false === getenv('SYMFONY_PATCH_TYPE_DECLARATIONS')) {
+    putenv('SYMFONY_PATCH_TYPE_DECLARATIONS=force=2');
+    echo 'Patching type declarations in all methods';
+}
+
+if (is_file($autoload = __DIR__.'/../../../autoload.php')) {
+    // noop
+} elseif (is_file($autoload = __DIR__.'/../../../../../vendor/autoload.php')) {
+    // noop
+} else {
+    echo 'Cannot find the Composer autoloader, did you forget to run "composer install"?';
+    exit(1);
+}
+
+if (is_file($phpunitAutoload = pathinfo($autoload, \PATHINFO_DIRNAME).'/bin/.phpunit/phpunit/vendor/autoload.php')) {
+    require $phpunitAutoload;
+}
+
+$loader = require $autoload;
+
+Symfony\Component\ErrorHandler\DebugClassLoader::enable();
+
+$exclude = getenv('SYMFONY_PATCH_TYPE_EXCLUDE') ?: '/'.preg_quote(\DIRECTORY_SEPARATOR.'vendor'.\DIRECTORY_SEPARATOR).'/';
```

Comment:
It'd be great to document `SYMFONY_PATCH_TYPE_EXCLUDE` right into this file, maybe the rest a bit too.

Reply:
note that vendor files are already ignored by `DebugClassLoader::checkClasses()`, see
https://github.com/symfony/symfony/blob/64cbfd2569840777a54b849bb52a42aae084417c/src/Symfony/Component/ErrorHandler/DebugClassLoader.php#L532